File: C++/Problems/Problems/problem_2855.cpp
#include "problem_2855.h"

bool problem_2855::test() {
    std::vector nums { 3, 4, 5, 1, 2 };

    int expected = 2;

    auto result = minimumRightShifts(nums);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_2855::minimumRightShifts(std::vector<int> &nums) {
    int n = nums.size();
    int pos = nums[n - 1] >= nums[0] ? n - 1 : -1;
    for (int i = 0; i < n - 1; ++i)
    {
        if (nums[i] > nums[i + 1])
        {
            if (pos != -1)
                return -1;
            pos = i;
        }
    }

    return n - pos - 1;
}

File: Python/Problems/Problems/problem2855.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem


class Problem2855(Problem):
    def test(self) -> bool:
        nums = [3, 4, 5, 1, 2]

        expected = 2

        result = self.minimumRightShifts(nums)

        print(result)

        return result == expected

    def minimumRightShifts(self, nums: List[int]) -> int:
        n = len(nums)
        pos = n - 1 if nums[n - 1] >= nums[0] else -1
        for i in range(n - 1):
            if nums[i] > nums[i + 1]:
                if pos != -1:
                    return -1
                pos = i

        return n - pos - 1